### PR TITLE
JS(extractor): improve parser support for flowtype syntax

### PR DESCRIPTION
**Pull request by esben-semmle**
A port of the approved https://git.semmle.com/Semmle/code/pull/28621 to `/ql`.

### CODE REVIEW

File: javascript/extractor/src/com/semmle/jcorn/flow/FlowParser.java [hunk 1]
```diff
@@ -223,14 +227,29 @@ private void flowParseDeclareModule(Position start) {
...
+				this.flowParseDeclare(stmtStart);
+			}
 		}
 		this.expect(TokenType.braceR);
 	}
 
+	private void flowParseDeclareImport(Position stmtStart) {
+		String kind = flowParseImportSpecifiers();
+		if (kind == null) {
+			this.raise(stmtStart, "Imports within a `declare module` body must always be `import type` or `import typeof`!");
```
Comment (xiemaisi) [inline javascript/extractor/src/com/semmle/jcorn/flow/FlowParser.java]: We don't normally use exclamation marks in error messages, I think.

File: javascript/extractor/tests/esnext/input/nullish-coalescing.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+x1 ?? y1;
```
Comment (xiemaisi) [inline javascript/extractor/tests/esnext/input/nullish-coalescing.js]: Unrelated?

**Comment by esben-semmle** [comment]
Fixed x2, and squashed.
